src/index.d.ts

```diff
@@ -0,0 +1,43 @@
+
+declare module 'react-hookstore' {
+    type StateCallback<TState> = (state: TState) => void;
```

Comment:
What is TState? I don't see it declared anywhere, and after googling (it could be a built-in type) I couldnt find anything

Reply:
Sure, TState is a [generic type parameter](https://www.typescriptlang.org/docs/handbook/generics.html), and is part of what makes TypeScript so powerful. So in the `StateCallback<TState>` example, if you have a method that takes `StateCallback<TState>` and you pass into it a function that is of type `(state: number) => void`, then the type becomes `StateCallback<number>`. That allows `TState` to permeate all the way down, so that you have strong typing. Without generics, state would be typed to `any` and you'd have none of the benefits of TypeScript. In this case I called it `TState` to be clear about what the type parameter is, but it could be called anything.

Reply:
Said another way, `StateCallback<number>` becomes `(state: number) => void`, while `StateCallback<MyComplexStateType>` becomes `(state: MyComplexStateType) => void`

Reply:
Ok, Typescript is amazing. I wanted to start learning it for so long now and I think its finally time.

Reply:
Yes it is 😄 It would be even better if the library is written in it, because then it would generate the typings file for you and you'd have compiler guarantees about type safety, but providing a typings file alongside index.js is just a step towards that, and allows TypeScript users of this library to have type safety.

Reply:
The only think I'd like to see changed in this PR is the tab space (2 instead of 4) I guess. Otherwise I want to merge it